v2/internal/genericarmclient/generic_client.go

```diff
@@ -59,14 +64,19 @@ func NewGenericClient(cloudCfg cloud.Configuration, creds azcore.TokenCredential
 		options = &GenericClientOptions{}
 	}
 
+	ua := options.UserAgent
```

Comment:
Nice. If you wanted to you could pull the actual useragent out to the controller from genericClient useragent_policy.

Reply:
Sorry, I don't understand the suggestion. Will follow up offline.

Reply:
Basically, if you look at where the existing `userAgent` field is defined, it's defined in `v2/internal/genericarmclient/useragent_policy.go` as `var userAgent = "aso-controller/" + version.BuildVersion`

But, since `genericarmclient` is supposed to be generic, that feels like the wrong place for the controllers constant. Instead, really the controller useragent should be done like the asoctl useragent, passed in as a parameter on `options`, rather than genericarmclient assuming it's being used in the controller.